src/hotspot/os/linux/os_linux.cpp

```diff
@@ -4746,19 +4746,46 @@ int os::active_processor_count() {
   return active_cpus;
 }
 
+static volatile int warn_invalid_processor_id = 1;
```

Comment:
Maybe moving this var into the function, since it's only used inside it.

Reply:
Doing so will come with the cost of always having to run a pthread_once() in function entry.

Reply:
I'm was wrong here. Since the initialization if effectively const/constexp, there will not be any "pthread_once" overhead here. Moved the static variable inside the function in the latest commit.